### DISCUSSION TITLE
How to add custom code in story tab

### DISCUSSION THREAD

**Discussion by NishargShah** [storybookjs/storybook#13728] [Help] [answered]
I am making documentation of one package with the help of `storybook` and I already completed it but I need to show the code of that example, so how can I show the code to the users so they can copy-paste the whole code and run the code easily on their local machine

I searched on google and I got a package called `@storybook/addon-storysource`, it's nice but it took the only dynamic code but I don't want to show that so I did R & D, and I got a reference of `storiesOf` API but that shows my whole code of file even it shows `storybook` code also

```
import React from 'react';
import { storiesOf } from '@storybook/react';
import Mapper from './components/Mapper';

const SimpleApp = {
  title: 'Examples/Simple',
  component: Mapper,
  parameters: {
    controls: { hideNoControlsWarning: true },
  },
};

const TopComponent = () => (
  <div className="top_container">
    <h1 className="title">Simple Example</h1>
  </div>
);

export const Simple = () => <Mapper TopComponent={TopComponent} />;

storiesOf('Simple1', module).add('Simple', Simple);

export default SimpleApp;
```

Can someone help me to add specific code in the `story tab` like

```
const URL = 'https://google.com';
const MAP = [];

<Mapper 
   src={URL} 
   map={MAP}
/>
```

**Comment by NishargShah** [comment]
any update??

**Comment by NishargShah** [comment, marked as answer]
I made my own addon of that and successfully completed my goal

**Reply by hectormartinez-facephi** [reply]
Could you share your addon? Thank you!

**Reply by NishargShah** [reply]
@hectormartinez-facephi Please check https://github.com/img-mapper/react-docs
